elyra/cli/pipeline_app.py

```diff
@@ -0,0 +1,172 @@
+#
+# Copyright 2018-2020 Elyra Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import asyncio
+import click
+import json
+import os
+import requests
+
+from yaspin import yaspin
+from colorama import Fore, Style
+
+from elyra import __version__
+from elyra.pipeline import PipelineParser, PipelineProcessorManager
+from jupyter_server.serverapp import list_running_servers
+
+
+def _preprocess_pipeline(pipeline_path, runtime, runtime_config, work_dir):
+    pipeline_path = os.path.expanduser(pipeline_path)
+    working_dir = os.path.expanduser(work_dir)
+    pipeline_abs_path = os.path.join(working_dir, pipeline_path)
+    pipeline_dir = os.path.dirname(pipeline_abs_path)
+    pipeline_name = os.path.splitext(os.path.basename(pipeline_abs_path))[0]
+
+    with open(pipeline_abs_path) as f:
+        pipeline_definition = json.load(f)
+
```

Comment:
We do pipeline validation in `PipelineParser().parse(...)` but that happens after this preprocess function. If the pipeline is invalid I would assume this would crash before making it to the nicer error messages in `parse`. 

We should also check the pipeline version, I don’t think I saw logic for that in `parse`